### PR TITLE
[MAINTENANCE] Cleanup Metrics and ExecutionEngine methods

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
Make stylistic improvements and remove unused code:

* Remove misleading code in implementation of `ExecutionEngine.resolve_metric_bundle()` in `SqlAlchemyExecutionEngine` and in `SparkDFExecutionEngine` classes and change variables to improve code readability.
* Remove unused code in `MapMetricProvider` and improve code readability.
* Update type hints in test modules.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-1347/GREAT-1366
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/execution_engine/execution_engine.py [hunk 1]
```diff
@@ -416,10 +416,9 @@ def resolve_metrics(  # noqa: C901 - 16
         if len(metric_fn_bundle) > 0:
             try:
                 # an engine-specific way of computing metrics together
-                # NOTE: DH 20220328: This is where we can introduce the Batch Metrics Store (BMS)
```
Comment (alexsherstinsky) [inline great_expectations/execution_engine/execution_engine.py:419]: @donaldheppner Please note that this location will not work for BMS as sometimes partially-computed metrics are produced here.  A better candidate would be surrounding the call to `ValidationGraph.MetricsCalculator.resolve_validation_graph()` -- happy to discuss.  Thank you.

File: great_expectations/execution_engine/sparkdf_execution_engine.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 import copy
 import datetime
 import logging
```
Comment (Kilo59) [inline great_expectations/execution_engine/sparkdf_execution_engine.py:5]: `sparkdf_execution_engine.py` is currently not being type-checked but it only has a few issues.
Can you remove it from the `pyproject.toml` file and deal with any issues?
Happy to help with any typing questions.

https://github.com/great-expectations/great_expectations/blob/develop/pyproject.toml#L63

Comment (alexsherstinsky) [reply great_expectations/execution_engine/sparkdf_execution_engine.py:5]: @Kilo59 Good idea -- woking on it -- will let you know when to re-review!  Thanks!

File: great_expectations/execution_engine/sparkdf_execution_engine.py [hunk 3]
```diff
@@ -293,17 +298,15 @@ def get_batch_data_and_markers(
...
+            reader_method = batch_spec.reader_method
+            reader_options = batch_spec.reader_options or {}
+            path = batch_spec.path
+            schema = reader_options.get("schema")
 
             # schema can be a dict if it has been through serialization step,
             # either as part of the datasource configuration, or checkpoint config
             if isinstance(schema, dict):
-                schema: pyspark.sql.types.StructType = sparktypes.StructType.fromJson(
+                schema = pyspark.sql.types.StructType = sparktypes.StructType.fromJson(
```
Comment (Kilo59) [inline great_expectations/execution_engine/sparkdf_execution_engine.py]:
```suggestion
                schema = sparktypes.StructType.fromJson(
```
Is this a mistake?

**Review by Kilo59 — APPROVED** [review]

**Review by Kilo59 — CHANGES_REQUESTED** [review]
See comment.
Can you explain why we are doing this?
Other than that I have no issues.

https://github.com/great-expectations/great_expectations/pull/6371/files#r1024014250

File: great_expectations/execution_engine/sparkdf_execution_engine.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sparkdf_execution_engine.py]: @Kilo59 Thank you for spotting this pesky typo!  Fixed -- committing shortly.

**Review by Kilo59 — APPROVED** [review]

**Comment by alexsherstinsky** [comment]
> https://github.com/great-expectations/great_expectations/pull/6371/files#r1024014250

@Kilo59 To make sure I do not miss it -- if still relevant -- I cannot find the exact item your question refers to.  Please clarify. 
 Thanks.

File: great_expectations/execution_engine/execution_engine.py [hunk 4]
```diff
@@ -41,7 +41,7 @@
 from great_expectations.validator.metric_configuration import MetricConfiguration
 
 if TYPE_CHECKING:
-    from great_expectations.expectations.metrics import MetricProvider
+    from great_expectations.expectations.metrics.metric_provider import MetricProvider
```
Comment (alexsherstinsky) [inline great_expectations/execution_engine/execution_engine.py:44]: @Kilo59 if this change of import is the focus of the question, then the answer is that previously, the import was not leading to a resolvable reference, causing numerous errors in Azure logs.  Thanks.

**Pull request merged**